docs/application/native/guides/text-input/capi-ui-gesture.md

```diff
@@ -0,0 +1,250 @@
+  The candidates of the gesture types are defined in the `gesture_common.h` header file as the enumerations of `hand_gesture_type_e`:
+
+    ```c
+    void
+    is_supported_gesture_type(hand_gesture_h gesture_h, hand_gesture_type_e gesture_type, bool* is_supported)
+    {
+        int ret;
+        ret = hand_gesture_is_supported_type(gesture_h, gesture_type, is_supported);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+- Get the information of the Gesture engine service used in current.
+
+  You can get the application ID and name of the current Gesture engine service using `hand_gesture_get_engine_info()`:
+
+    ```c
+    void
+    get_info(hand_gesture_h gesture_h)
+    {
+        int ret;
+        char* engine_app_id;
+        char* engine_name;
+
+        ret = hand_gesture_get_engine_info(gesture_h, &engine_app_id, &engine_name);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="set_option"></a>
+## Set option
+
+You can select an option to detect hand gestures continuously or not. If you want to make the application detect hand gestures continuously, set the option as `HAND_GESTURE_OPTION_ALWAYS_ON` using `hand_gesture_set_option()`:
+
+```c
+void
+set_option(hand_gesture_h gesture_h)
+{
+    int ret;
+    ret = hand_gesture_set_option(gesture_h, HAND_GESTURE_OPTION_ALWAYS_ON);
+    if (HAND_GESTURE_ERROR_NONE != ret)
+        /* Error handling */
+}
+```
+
+
+<a name="callback"></a>
+## Set and unset callbacks
+
+The Gesture provides two types of callbacks for getting error and gesture recognition result.
+The enum values, as well as the parameter details, for the callback parameters are defined in the `gesture.h` and `gesture_common.h` header files.
+
+- Error callback
+
+  When an error occurs, the Gesture library sends a message using a callback:
+
+  > [!NOTE]
+  > Ensure that you set the error callback function before calling `hand_gesture_start_recognition()`.
+
+    ```c
+    /* Callback */
+    void
+    error_cb(hand_gesture_h gesture_h, hand_gesture_error_e reason, const char* msg, void* user_data)
+    {
+        /* Your code */
+    }
+
+    /* Set */
+    void
+    set_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_set_error_cb(gesture_h, error_cb, NULL);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+    /* Unset */
+    void
+    unset_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_unset_error_cb(gesture_h);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+- Recognition result
+
+  When the hand gesture is recognized, the recognized result is delivered through `hand_gesture_recognition_cb()`.
+
+  You can [set the recognition result callback function using `hand_gesture_start_recognition()`](#start_stop).
```

Comment:
How about applying the hyperlink to the core point?

```suggestion
  You can [set the recognition result callback function](#start_stop) using `hand_gesture_start_recognition()`.
```

Reply:
Please incorporate thsi comment:

https://github.com/Samsung/tizen-docs/pull/1280/files#r508235524

Reply:
I changed this sentence. Please review this again.